Content.Server/Maps/GameMapManager.cs

```diff
@@ -177,22 +181,28 @@ public int GetMapQueuePriority(string gameMapProtoName)
                 return i;
             i++;
         }
-
         return _mapQueueDepth;
     }
 
-    public GameMapPrototype? SelectMapInQueue()
+    private GameMapPrototype GetFirstInRotationQueue()
     {
-        Logger.InfoS("mapsel", string.Join(", ", _previousMaps));
+        Logger.InfoS("mapsel", $"map queue: {string.Join(", ", _previousMaps)}");
+
         var eligible = CurrentlyEligibleMaps()
-            .Select(x => (proto: x, weight: GetMapQueuePriority(x.ID)))
-            .OrderByDescending(x => x.weight).ToArray();
-        Logger.InfoS("mapsel", string.Join(", ", eligible.Select(x => (x.proto.ID, x.weight))));
+            .Select(x => (proto: x, weight: GetMapRotationQueuePriority(x.ID)))
+            .OrderByDescending(x => x.weight)
+            .ToArray();
+
+        Logger.InfoS("mapsel", $"eligible queue: {string.Join(", ", eligible.Select(x => (x.proto.ID, x.weight)))}");
+        
         if (eligible.Length is 0)
             return null;
```

Comment:
So I updated this to master but is there even any guarantee of an eligible map.

Reply:
If the YML is properly configured then `CurrentlyEligibleMaps()` should always return at least one map. If it dosen't pull any maps with regular conditions then it currently falls back to Saltern (`Fallback: true`)